### PR TITLE
Falls Creek

### PR DESCRIPTION
**Pull request by OkiRules**
Hi,

I think I have done everything properly expect for the index.js part. I'm still not sure how to do this... Could you please leave some helpful information on how to do this, so I can add resort without your help...

Thanks in advance

### CODE REVIEW

File: lib/resorts/falls-creek/resort.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    "host": "https://www.fallscreek.com.au",
+    "pathname": "/snowreport/"
+  },
+  "tags": [
+    "Australia",
+    "Victoria"
+  ],
+  "ll": [
+    147.287,
+    36.8659
```
Comment (pirxpilot) [inline lib/resorts/falls-creek/resort.json]: unless you guys moved Australia overnight latitude should be a negative number ;-)

File: lib/resorts/falls-creek/index.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+module.exports = {
+  selector: 'div[class=="snow-inner-table-content"]',
```
Comment (pirxpilot) [inline lib/resorts/falls-creek/index.js]: '.snow-table-lift-status .snow-inner-table-content .table-row' selects every row in the table and from that you can try finding children with name and status
That said - this is again the case HTML does not have status information. Looks like you need to parse https://www.fallscreek.com.au/wp-content/uploads/FCSnowReport.json to get it. Use Mount Hotham as an example of how to do it - you'll have to change both 'index.js' and the test file.

**Comment by OkiRules** [comment]
Ok thanks. I’ll try have a look today. One thing though, how did you work out that the falls creek data comes from the json file link?

Thanks,
Oscar

**Comment by OkiRules** [comment]
Ok ive added some stuff but still cant work out how to parse the json?

**Comment by pirxpilot** [comment]
> Ok thanks. I’ll try have a look today. One thing though, how did you work out that the falls creek data comes from the json file link? Thanks, Oscar

I usually start by opening the downloaded HTML file (it's in test/examples directory). I try to search for the name of the lift that is visible on the page. If I can find it I try to see if there is a status name or status icon/image next to it. In this case there was not. Next I open the page in the browser (I use Chrome). I open the Developer Tools (either Tools -> Developer Tools or Shift-Ctrl-I and I open the Network panel. I refresh the page and see what requests it makes to the server. XHR tab is usually worth a look - sometimes URL of the request is pretty obvious - but even if it is not you can click on each request and see the preview of the response. Try it with couple of resorts.
BTW: developer tools are pretty helpful if status is in HTML as well. You can highlight the status element and right-click on it and choose Copy->Selector from the popup menu. It's rarely the best selector to use in liftie but you can pair it down to what you need to use in `index.js` file.

**Comment by OkiRules** [comment]
Ok thanks. I tried to work out how to parse the Jason but could not succeed as it is different to mt Hotham... please take a look...

Thanks,
Oscar

**Comment by pirxpilot** [comment]
> Ok ive added some stuff but still cant work out how to parse the json?

You need to add falls-creek.json to the examples, and you need to rewrite the test to follow the Mt Hotham example. There are quite a few resorts in liftie that get status info from JSON. Search for "api" key in resort.json to see which ones.

**Comment by OkiRules** [comment]
Ok... I changed the js files to as follows:

`
var parse = require('../../lib/lifts/parse')('falls-creek');

/*global describe, it */
describe('parse falls-creek', function() {

  it('should return lift status', function() {
    var data = require('./example/falls-creek.json');
    var expected = {
      	'Halleys Comet': 'closed',
        'Towers Chair': 'closed',
        'Drovers Dream Chair': 'closed',
        'Board Walk': 'closed',
        'Mouse Trap': 'closed',
        'Monkey Bars': 'closed',
        'Petes Train': 'closed',
        'Gully Chair': 'closed',
        'Eagle Express': 'closed',
        'Scotts Chair': 'closed',
        'Ruined Castle Chair': 'closed',
        'Summit Chair': 'closed',
        'Lakeside Poma': 'closed',
        'International Poma': 'closed'
    };
    parse(data).should.eql(expected);
  });
});

`

but then while running `make` I got this error:

`MacBook-Pro:liftie somethingMacbookPro$ make
./node_modules/.bin/jshint app.js bin/generate lib test
./node_modules/.bin/mocha --recursive --require should
internal/modules/cjs/loader.js:583
    throw err;
    ^

Error: Cannot find module 'lib/lifts/parse'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:581:15)
    at Function.Module._load (internal/modules/cjs/loader.js:507:25)
    at Module.require (internal/modules/cjs/loader.js:637:17)
    at require (internal/modules/cjs/helpers.js:22:18)
    at Object.<anonymous> (/Users/OkiRules/liftie/test/resorts/falls-creek.js:1:75)
    at Module._compile (internal/modules/cjs/loader.js:689:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:700:10)
    at Module.load (internal/modules/cjs/loader.js:599:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:538:12)
    at Function.Module._load (internal/modules/cjs/loader.js:530:3)
    at Module.require (internal/modules/cjs/loader.js:637:17)
    at require (internal/modules/cjs/helpers.js:22:18)
    at /Users/OkiRules/liftie/node_modules/mocha/lib/mocha.js:250:27
    at Array.forEach (<anonymous>)
    at Mocha.loadFiles (/Users/OkiRules/liftie/node_modules/mocha/lib/mocha.js:247:14)
    at Mocha.run (/Users/OkiRules/liftie/node_modules/mocha/lib/mocha.js:576:10)
    at Object.<anonymous> (/Users/OkiRules/liftie/node_modules/mocha/bin/_mocha:637:18)
    at Module._compile (internal/modules/cjs/loader.js:689:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:700:10)
    at Module.load (internal/modules/cjs/loader.js:599:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:538:12)
    at Function.Module._load (internal/modules/cjs/loader.js:530:3)
    at Function.Module.runMain (internal/modules/cjs/loader.js:742:12)
    at startup (internal/bootstrap/node.js:282:19)
    at bootstrapNodeJSCore (internal/bootstrap/node.js:743:3)
make: *** [test] Error 1
MacBook-Pro:liftie somethingMacbookPro$ 
`

Any suggestions?

**Comment by OkiRules** [comment]
Please help me parse the JSON, I really need help.

**Pull request closed without merging**